state-chain/pallets/cf-elections/src/electoral_systems/tests/block_witnesser.rs

```diff
@@ -603,48 +555,44 @@ fn elections_resolved_out_of_order_has_no_impact() {
 		)
 		.expect_consensus_multi(vec![
 			(
-				// no consensus
-				generate_votes((0..20).collect(), (0..20).collect(), Default::default(), vec![]),
+				generate_votes(
+					(0..20).collect(),
+					(0..20).collect(),
+					Default::default(),
+					vec![],
+					None,
+				),
 				None,
 			),
 			(
-				// consensus
 				generate_votes(
 					(0..40).collect(),
 					Default::default(),
 					Default::default(),
 					vec![1, 3, 4],
+					None,
 				),
-				Some(vec![1, 3, 4]),
+				Some((vec![1, 3, 4], None)),
 			),
 		])
 		// no progress on external chain but on finalize called again
-		// TODO: Check the new elections have kicked off correct
 		.test_on_finalize(
-			&vec![ChainProgress::Range(range_n(
-				INIT_LAST_BLOCK_RECEIVED + 1 + (NUMBER_OF_ELECTIONS as u64),
-				1,
-			))],
+			&vec![None],
 			|pre_state| {
 				assert_eq!(
 					pre_state.unsynchronised_state.elections.ongoing.len(),
 					NUMBER_OF_ELECTIONS as usize
 				);
 			},
 			vec![
-				// one extra election created
-				// TODO: this should check that we actually called properties for the newly created
-				// election. as things are we cannot see whether we closed an election and
-				// opened a new one or simply kept the previously ongoing ones
+				// one extra election created for the block that we didn't reach consensus over
 				Check::<SimpleBlockWitnesser>::generate_election_properties_called_n_times(
-					NUMBER_OF_ELECTIONS as u8,
+					NUMBER_OF_ELECTIONS.saturating_sub(1) as u8,
 				),
 				// we should have resolved one election, and started one election
-				Check::<SimpleBlockWitnesser>::number_of_open_elections_is(2),
-				// Check::<SimpleBlockWitnesser>::unprocessed_data_is(vec![(
-				// 	SECOND_ELECTION_BLOCK_CREATED,
-				// 	vec![1, 3, 4],
-				// )]),
+				Check::<SimpleBlockWitnesser>::number_of_open_elections_is(
+					NUMBER_OF_ELECTIONS.saturating_sub(1),
+				),
```

Comment:
Actually: `// we should have resolved one election, and so there's one less ongoing`

We should also check that we have emitted the prewitness events for the resolved block. And also for the next block in the next on_finalize.